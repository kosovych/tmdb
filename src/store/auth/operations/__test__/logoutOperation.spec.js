import { logoutOperation } from '../logoutOperation'

describe('logoutOperation()', () => {
  it('has valid attributes', () => {
    expect(logoutOperation).toMatchSnapshot()
  })
  it('calls localStorage.removeItem() with right arguments', () => {
    const done = jest.fn()
    const spy = jest.spyOn(Storage.prototype, 'removeItem')
    logoutOperation.process(null, null, done)

    expect(spy).toHaveBeenCalledTimes(2)

    expect(spy).toHaveBeenNthCalledWith(1, 'sessionId')
    expect(spy).toHaveBeenNthCalledWith(2, 'userId')
  })
})
