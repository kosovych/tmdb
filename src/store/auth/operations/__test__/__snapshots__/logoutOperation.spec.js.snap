// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logoutOperation() has valid attributes 1`] = `
Object {
  "cancelType": undefined,
  "debounce": 0,
  "latest": true,
  "name": undefined,
  "process": [Function],
  "processOptions": Object {
    "dispatchMultiple": true,
  },
  "throttle": 0,
  "transform": undefined,
  "type": "auth/LOGIN_LOGOUT",
  "validate": undefined,
  "warnTimeout": 60000,
}
`;
